pkg/sources/fake/fake.go

```diff
@@ -15,24 +15,19 @@ const (
 
 var startTime = time.Now()
 
-func getBattery(id uint16) msgs.TelemetryData {
+func getCAN(id uint16) msgs.CAN {
 	t := time.Since(startTime)
 	fid := float64(id)
-	voltage := uint16(33000 + 3000*math.Sin((t.Seconds()/20)+fid))
-	temperature := float32(33000 + 3000*math.Sin((t.Seconds()/20)+fid))
+	data := uint16(33000 + 3000*math.Sin((t.Seconds()/20)+fid))
 
-	return msgs.NewTelemetryData(&msgs.BatteryModule{
-		ID:          id,
-		Voltage:     voltage,
-		Temperature: temperature,
-	})
+	return msgs.NewCAN(id, data)
 }
 
 // GenFake generates fake data
 func GenFake(bus *pubsub.MessageBus) {
 	for {
-		for battery := uint16(1); battery <= 36; battery++ {
-			bus.Publish("battery", getBattery(battery))
+		for msg := uint16(1); msg <= 36; msg++ {
+			bus.Publish("CAN", getCAN(msg))
```

Comment:
Correct I think that is exactly what we want. What I was getting at is the publisher dispatches each msgs.CAN to different topics depending on its ID since we don't have discrete ID mappings for topics yet I just decided to have a single topic for now. Then the subscribers can just subscribe to topics they care about.